### DISCUSSION TITLE
next/link prefetch in Pages directory

### DISCUSSION THREAD

**Discussion by Fatima-Tahseen** [vercel/next.js#64914] [Help]
### Summary

I am in the middle of upgrading my nextjs app from version 12.2.4 to 14.
The issue I facing is that in version 14, the calls for the json data, sent on hover on the next/link are returning empty responses while they are returning actual page props on version 12.2.4

Another confusion is that according to [this](https://github.com/vercel/next.js/discussions/62937#discussioncomment-8695692) the call for json data shouldnt even be sent but its being dispatched and returns empty response.

**Comment by mogulist** [comment]
I am experiencing almost the same issue with latest next.js version. 
I found that x-middleware-prefetch header should be remove in order to prefetch respond correct data. Otherwise, prefetch requests don't reach `getServerSideProps()`. Following is my solution:
```
export async function middleware(req) {
  const requestHeaders = new Headers(req.headers);
  requestHeaders.delete("x-middleware-prefetch");

  return NextResponse.next({
    request: {
      headers: requestHeaders,
    },
  });
}
```
